### ISSUE TITLE
Run Configuration: Project selection without effect

### ISSUE THREAD

**Issue by hosswald** [GoogleCloudPlatform/google-cloud-intellij#2266]
- Cloud Tools for IntelliJ version: v18.8.1
- Google Cloud SDK version _(run `gcloud version` on the command line)_:
- OS: Windows 10 Version 1803

**What did you do?**
I created three different "Google Stackdriver Debug" run configs for three different projects. Refering to them as `project`, `project-dev` and `project-staging`.
I selected the Project in each config and it persists, and is even shared.
However, when I run the config, I am asked to "Attach to an Application" and the wrong Project is preselected, e.g. I run the correctly configured `project-staging` config and get a `project` preselection.

**What did you expect to see?**
Either there should be no popup at all (I would prefer to have an option to enable/disable this) or at least the correct project should be preselected.

**Comment by etanshaul** [comment]
Hi @hosswald, thanks for the report. 

@ivanporty I think what is happening here is that the "default cloud project" logic that we added is causing each of @hosswald's Stackdriver run configs to override the next one. i.e. each time a new run config is created and a project selected, it then sets itself as the default cloud project, loosing the value in the previously configured run configs.

My intuition here is that we should not override previously selected cloud projects with the "default" cloud project. Only if there is _no_ selection then it would make sense to set it to the default project. @ivanporty wdyt?

**Comment by ivanporty** [comment]
Right, I think there's an issue between `CloudAttachDialog` and `ProjectDebuggeeBinding` where the latter always pre-selects default Cloud project. I don't quite understand why the saved project is not set and where it's done, this needs more investigation on how Cloud debugger settings are applied.

**Comment by etanshaul** [comment]
sg. I added it to the 18.11.1 milestone for us to investigate. I can understand how this issue can be annoying so we should look into it. I assigned you since you have the default Cloud Project stuff more fresh in your mind, but feel free to unassign.

**Comment by hosswald** [comment]
@etanshaul much appreciated.

**Comment by etanshaul** [comment]
@hosswald thanks from brining this to our attention. It should be fixed.

@ivanporty can we update this to store the personalized state in a non-shared persistent state component, if that’s the right solution?

**Comment by etanshaul** [comment]
@hosswald just to make sure we understand your use-case properly.

You are sharing the Stackdriver Debugger run configuration and persisting it into VCS. Presumably you are sharing a GCP project with other teammates and would like to share the configuration, correct? Also, from your comment, I am assuming you have a project shared across multiple accounts.

@ivanporty the complication here is that since a GCP project is associated with an account, the teammates that this run configuration is being shared with would first have to select their own account for this shared project to appear and be selected. This seems to remove much value in sharing the run config in the first place (since the Stackdriver Debugger run config is limited to cloud project and username). 

Off the top of my head, one thing we could potentially consider doing is to share only the cloud project, and update the Cloud Project UI so that it prefills a Cloud Project when no user is provided, but shows it in an errored state until selecting an associated account.

**Comment by hosswald** [comment]
> @hosswald just to make sure we understand your use-case properly.
> 
> You are sharing the Stackdriver Debugger run configuration and persisting it into VCS. Presumably you are sharing a GCP project with other teammates and would like to share the configuration, correct? Also, from your comment, I am assuming you have a project shared across multiple accounts.
Exactly. Every team member accesses the same project, but of course with different Google accounts.

**Comment by etanshaul** [comment]
@hosswald we discussed this a bit here and are talking about some solutions, including potentially not persisting the username when the run configuration is shared. In the meantime I would suggest to unshare the run configuration and just have each user with his/her own configuration for now so that the username isn't stored (I'm guessing you may have already done this).

I was wondering: are you sharing any other run configurations (deployment, local run), or just Stackdriver debugger?

**Comment by hosswald** [comment]
@etanshaul currently we only share the Stackdriver run configurations. We deploy with a cloud service, so the deployment r.c. is irrelevant to us, and the local server does not seem to work with our gradle build script (we use a shared "appengineStart + attach 'remote' debugger" r.c. instead).

**Comment by hosswald** [comment]
I created a ticket for the above problem: https://github.com/GoogleCloudPlatform/google-cloud-intellij/issues/2282
